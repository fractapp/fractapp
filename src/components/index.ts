export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
export {TabMenu} from './TabMenu';
export {ChatShortInfo} from './ChatShortInfo';
export {PaymentMsg} from './PaymentMsg';
export {Contact} from './Contact';
export {Receiver} from './Receiver';
export {AmountInput} from './AmountInput';
export {AmountValue} from './AmountValue';
export {ReceiverWithEnterAddress} from './ReceiverWithEnterAddress';
