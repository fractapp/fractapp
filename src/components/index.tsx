export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
