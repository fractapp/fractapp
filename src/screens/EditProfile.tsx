import React, { useState } from 'react';
import { StyleSheet, View, Text, FlatList, Image, TouchableHighlight } from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import PhoneInput from "react-native-phone-number-input";

export const EditProfile = ({ navigation }: { navigation: any }) => {
    const Inputs = [
        {
            title: "Name",
            value: "Elshan Dzhafarov"
        },
        {
            title: "Username",
            value: "CryptoBadBoy"
        },
        {
            title: "Phone",
            value: "Write you phone"
        },
        {
            title: "Email",
            value: "Write your email"
        },
        {
            title: "Twitter",
            value: "Connect your twitter"
        }
    ]

    const renderItem = (item: any) => {
        const input = item.item
        return (
            <TouchableHighlight
                key={input.title}
                // onPress={() => navigation.navigate('EditProfile', { title: item.title })}
                underlayColor="#f8f9fb"
            >
                <View style={styles.input}>
                    <Text style={styles.title}>{input.title}</Text>
                    <Text style={styles.value}>{input.value}</Text>
                </View>
            </TouchableHighlight>
        )
    }
    return (
        <View style={styles.profile}>
            <View style={styles.avatar}>
                <MaterialCommunityIcons name="account-circle-outline" size={70} color="#2AB2E2" />
            </View>
            <View style={{ width: '90%' }}>
                <FlatList
                    showsVerticalScrollIndicator={false}
                    data={Inputs}
                    renderItem={renderItem}
                    keyExtractor={item => item.title}
                />
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    profile: {
        alignItems: 'center',
        justifyContent: 'center',
    },
    avatar: {
        marginTop: 20,
        width: 120,
        height: 120,
        borderColor: '#CCCCCC',
        borderWidth: 1,
        borderRadius: 60,
        alignItems: 'center',
        justifyContent: 'center',
    },
    input: {
        marginTop: 20,
        paddingBottom: 5,
        borderColor: '#CCCCCC',
        borderBottomWidth: 1,
    },
    title: {
        fontSize: 18,
        fontFamily: "RobotoRegular",
        fontStyle: "normal",
        fontWeight: "normal",
        color: "#888888",
    },
    value: {
        marginTop: 6,
        fontSize: 18,
        fontFamily: "RobotoRegular",
        fontStyle: "normal",
        fontWeight: "normal",
        color: "black",
    }
});