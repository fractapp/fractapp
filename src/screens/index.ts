export {ImportWallet} from './ImportWallet';
export {SaveSeed} from './SaveSeed';
export {SaveWallet} from './SaveWallet';
export {SettingWallet} from './SettingWallet';
export {Start} from './Start';
export {WalletDetails} from './WalletDetails';
export {WalletFileBackup} from './WalletFileBackup';
export {Wallets} from './Wallets';
export {Settings} from './Settings';
export {ConfirmSaveSeed} from './ConfirmSaveSeed';
export {ImportSeed} from './ImportSeed';
export {WalletFileImport} from './WalletFileImport';
export {GoogleDrivePicker} from './GoogleDrivePicker';
export {Receive} from './Receive';
export {Milestone} from './Milestone';
export {TransactionDetails} from './TransactionDetails';
export {SplashScreen} from './SplashScreen';
export {Backup} from './Backup';
export {NewPassCode} from './NewPassCode';
export {VerifyPassCode} from './VerifyPassCode';
export {EditProfile} from './EditProfile';
export {EditPhoneNumber} from './EditPhoneNumber';
export {SelectCountry} from './SelectCountry';
export {ConfirmCode} from './ConfirmCode';
export {EditEmail} from './EditEmail';
export {Navigation} from './Navigation';
export {Chats} from './Chats';
export {Chat} from './Chat';
export {Connecting} from './Connecting';
export {EditUsername} from './EditUsername';
export {EditName} from './EditName';
export {Search} from './Search';
export {SelectWallet} from './SelectWallet';
export {Send} from './Send';
export {EnterAmount} from './EnterAmount';
export {SendWithEnterAddress} from './SendWithEnterAddress';
