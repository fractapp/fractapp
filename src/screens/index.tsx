export { default as ImportWallet } from './ImportWallet';
export { default as SaveSeed } from './SaveSeed';

export { default as SaveWallet } from './SaveWallet';
export { default as SettingWallet } from './SettingWallet';
export { default as Start } from './Start';

export { default as WalletDetails } from './WalletDetails';

export { default as WalletEncryption } from './WalletEncryption';
export { default as Wallets } from './Wallets';
export { ConfirmSaveSeed } from './ConfirmSaveSeed'