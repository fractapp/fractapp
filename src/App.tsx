import React, {useState, useEffect, useContext} from 'react';
import SplashScreen from 'react-native-splash-screen';
import {View, Alert, StatusBar} from 'react-native';
import {Dialog, PassCode} from 'components';
import tasks from 'utils/tasks';
import GlobalStore from 'storage/Global';
import DialogStore from 'storage/Dialog';
import {Navigation} from 'screens';
import changeNavigationBarColor, {
  hideNavigationBar,
  showNavigationBar,
} from 'react-native-navigation-bar-color';
import PasscodeUtil from 'utils/passcode';
import {showMessage} from 'react-native-flash-message';
import DB from 'storage/DB';
import {useNetInfo} from '@react-native-community/netinfo';
import AccountsStore from 'storage/Accounts';
import PricesStore from 'storage/Prices';
import ChatsStore from 'storage/Chats';
import TransactionsStore from 'storage/Transactions';

export default function App() {
  const globalContext = useContext(GlobalStore.Context);
  const dialogContext = useContext(DialogStore.Context);
  const accountsContext = useContext(AccountsStore.Context);
  const pricesContext = useContext(PricesStore.Context);
  const chatsContext = useContext(ChatsStore.Context);
  const transactionsContext = useContext(TransactionsStore.Context);

  const [isLoading, setLoading] = useState<Boolean>(false);
  const [isLocked, setLocked] = useState<Boolean>(false);
  const [isConnected, setConnected] = useState<Boolean>(true);
  const netInfo = useNetInfo();

  const onSubmitPasscode = async (passcode: Array<number>) => {
    let hash = await DB.getPasscodeHash();
    let salt = await DB.getSalt();

    if (salt == null) {
      Alert.alert('Please contact support: support@fractapp.com');
      return;
    }

    if (hash === PasscodeUtil.hash(passcode.join(''), salt)) {
      setLocked(false);
    } else {
      showMessage({
        message: 'Incorrect passcode',
        type: 'danger',
        icon: 'warning',
      });
    }
  };

  useEffect(() => {
    if (isLoading) {
      return;
    }
    console.log('start ' + new Date().toTimeString());
    hideNavigationBar();
    setLoading(true);

    DB.isAuthed().then((isAuthed) => {
      if (!isAuthed) {
        showNavigationBar();
        setLoading(false);
        changeNavigationBarColor('#FFFFFF', true, true);
        SplashScreen.hide();

        console.log('end ' + new Date().toTimeString());
        return;
      }

      tasks.init(
        globalContext,
        accountsContext,
        pricesContext,
        chatsContext,
        transactionsContext,
      );
    });
  }, [globalContext.state.isAuthed]);
  useEffect(() => {
    if (!globalContext.state.isInitialized) {
      return;
    }

    setLocked(globalContext.state.isPasscode);
    tasks
      .createTask(
        globalContext.state.isSynced,
        accountsContext,
        pricesContext,
        globalContext,
        chatsContext,
        transactionsContext,
      )
      .then(() => {
        showNavigationBar();
        setLoading(false);
        changeNavigationBarColor('#FFFFFF', true, true);
        SplashScreen.hide();
        console.log('end ' + new Date().toTimeString());
      });
  }, [globalContext.state.isInitialized]);
  useEffect(() => {
    if (!globalContext.state.isInitialized) {
      return;
    }

    if (netInfo.isConnected && !isConnected) {
      showMessage({
        message: 'Connection restored',
        type: 'success',
        icon: 'success',
        position: 'top',
      });
    }
    if (!netInfo.isConnected) {
      showMessage({
        message: 'Invalid connection',
        type: 'danger',
        icon: 'danger',
        hideOnPress: false,
        autoHide: false,
        position: 'top',
      });
    }

    setConnected(netInfo.isConnected);
  }, [netInfo.isConnected]);

  if (isLoading) {
    return <View />;
  }

  return (
    <View
      style={{
        flex: 1,
      }}>
      <StatusBar
        backgroundColor={'white'}
        barStyle={'dark-content'}
        hidden={false}
      />
      {isLocked ? (
        <PassCode
          isBiometry={globalContext.state.isBiometry}
          isBiometryStart={globalContext.state.isBiometry}
          description={'Enter passcode'}
          onSubmit={onSubmitPasscode}
        />
      ) : (
        <Navigation isInitialized={globalContext.state.isInitialized} />
      )}
      <Dialog
        visible={dialogContext.state.visible}
        onPress={dialogContext.state.onPress}
        title={dialogContext.state.title}
        text={dialogContext.state.text}
      />
    </View>
  );
}
