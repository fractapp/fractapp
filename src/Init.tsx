import React, { useEffect } from 'react';
import { Alert, Dimensions, StatusBar, View } from 'react-native';
import tasks from 'utils/tasks';
import GlobalStore from 'storage/Global';
import AccountsStore from 'storage/Accounts';
import { useDispatch, useSelector } from 'react-redux';
import UsersStore from 'storage/Users';
import ChatsStore from 'storage/Chats';
import ServerInfoStore from 'storage/ServerInfo';
import { Store } from 'redux';
import { Loader } from 'components/Loader';
import { Adaptors } from 'adaptors/adaptor';
import SplashScreen from 'react-native-splash-screen';
import StringUtils from 'utils/string';

const Init = ({store}: {store: Store}) => {
  const dispatch = useDispatch();
  const globalState: GlobalStore.State = useSelector((state: any) => state.global);
  const usersState: UsersStore.State = useSelector((state: any) => state.users);
  const chatsState: ChatsStore.State = useSelector((state: any) => state.chats);
  const accountsState: AccountsStore.State = useSelector((state: any) => state.accounts);
  const serverInfoState: ServerInfoStore.State = useSelector((state: any) => state.serverInfo);

  useEffect(() => {
    if (globalState.loadInfo.isAllStatesLoaded) {
      return;
    }

    console.log('start ' + new Date().toTimeString());
  //  hideNavigationBar();

    (async () => {
      try {
        await Adaptors.init();
        await tasks.init(dispatch);
      } catch (e) {
        Alert.alert(StringUtils.texts.showMsg.invalidConnection);
      }
    })();
  }, []);

  useEffect(() => {
    console.log('Is Global init? ' + globalState.isInitialized);
    console.log('Is Accounts init? ' + accountsState.isInitialized);
    console.log('Is Users init? ' + usersState.isInitialized);
    console.log('Is Chats init? ' + chatsState.isInitialized);
    console.log('Is Server Info init? ' + serverInfoState.isInitialized);

    (async () => {
      if (
        !globalState.isInitialized ||
        !accountsState.isInitialized ||
        !usersState.isInitialized ||
        !chatsState.isInitialized ||
        !serverInfoState.isInitialized
      ) {
        return;
      }

      dispatch(GlobalStore.actions.setAllStatesLoaded(true));
    })();
  }, [
    globalState.isInitialized,
    accountsState.isInitialized,
    usersState.isInitialized,
    chatsState.isInitialized,
    serverInfoState.isInitialized,

    globalState.authInfo.hasWallet,
  ]);

  return (
    <View
      style={{
        flex: 1,
      }}>
      <StatusBar
        backgroundColor={'white'}
        barStyle={'dark-content'}
        hidden={false}
      />
      {globalState.loadInfo.isLoadingShow && (
        <View
          style={{
            display: 'flex',
            alignItems: 'stretch',
            position: 'absolute',
            backgroundColor: 'white',
            height: Dimensions.get('screen').height,
            width: '100%',
          }}>
          <Loader />
        </View>
      )}
    </View>
  );
};

export default Init;
