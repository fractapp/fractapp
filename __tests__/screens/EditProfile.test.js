import React, {useState, useContext} from 'react';
import renderer from 'react-test-renderer';
import {EditProfile} from 'screens/EditProfile';
import GlobalStore from 'storage/Global';
import {fireEvent, render} from '@testing-library/react-native';
import {launchImageLibrary} from 'react-native-image-picker/src/index';
import backend from 'utils/backend';
import StringUtils from 'utils/string';

jest.mock('storage/DB', () => ({}));
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
  useContext: jest.fn(),
  Linking: jest.fn(() => ({
    openURL: jest.fn(),
  })),
}));
jest.mock('utils/backend', () => ({
  getImgUrl: jest.fn(() => 'uri'),
  uploadAvatar: jest.fn(),
}));
jest.mock('react-native-image-picker/src/index', () => ({
  launchImageLibrary: jest.fn(),
}));

jest.mock('react-native-i18n', () => ({
  t: (value) => value,
}));


useState.mockImplementation((init) => [init, jest.fn()]);

it('Test view (empty)', () => {
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: jest.fn(),
  });
  const tree = renderer.create(<EditProfile navigation={null} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test view (full)', () => {
  const state = GlobalStore.initialState();
  state.profile = {
    id: 'id',
    name: 'name',
    username: 'username',
    phoneNumber: 'phoneNumber',
    email: 'email',
    isMigratory: false,
    avatarExt: 'png',
    lastUpdate: 123123,
  };
  useContext.mockReturnValueOnce({
    state: state,
    dispatch: jest.fn(),
  });

  const tree = renderer.create(<EditProfile navigation={null} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test clicks', () => {
  const state = GlobalStore.initialState();
  const nav = jest.fn();
  state.profile = {
    id: 'id',
    name: 'name',
    username: 'username',
    phoneNumber: '',
    email: '',
    isMigratory: false,
    avatarExt: 'png',
    lastUpdate: 123123,
  };
  useContext.mockReturnValueOnce({
    state: state,
    dispatch: jest.fn(),
  });

  const component = render(<EditProfile navigation={{navigate: nav}} />);
  fireEvent.press(component.getByText('name'));
  expect(nav).toBeCalledWith('EditName');

  fireEvent.press(component.getByText('@username'));
  expect(nav).toBeCalledWith('EditUsername');

  fireEvent.press(component.getByText(StringUtils.texts.edit.profile.phoneTitle));
  expect(nav).toBeCalledWith('EditPhoneNumber');

  fireEvent.press(component.getByText(StringUtils.texts.edit.profile.emailTitle));
  expect(nav).toBeCalledWith('EditEmail');
});

it('Test edit avatar', async () => {
  const state = GlobalStore.initialState();
  const nav = jest.fn();
  const dispatch = jest.fn();

  state.profile = {
    id: 'id',
    name: 'name',
    username: 'username',
    phoneNumber: '',
    email: '',
    isMigratory: false,
    avatarExt: 'png',
    lastUpdate: 123123,
  };
  useContext.mockReturnValueOnce({
    state: state,
    dispatch: dispatch,
  });

  const component = render(<EditProfile navigation={{navigate: nav}} />);
  await fireEvent.press(component.getByTestId('editAvatarBtn'));

  expect(launchImageLibrary.mock.calls[0][0]).toStrictEqual({
    mediaType: 'photo',
    includeBase64: true,
    maxWidth: 400,
    maxHeight: 400,
  });

  await launchImageLibrary.mock.calls[0][1]({base64: 'base64', type: 'type'});
  expect(dispatch).toBeCalledWith(GlobalStore.setLoading(true));
  expect(backend.uploadAvatar).toBeCalledWith('base64', 'type');
  expect(dispatch).toBeCalledWith(GlobalStore.setUpdatingProfile(true));
  expect(dispatch).toBeCalledWith(GlobalStore.setLoading(false));
});

it('Test edit avatar (negative)', async () => {
  const state = GlobalStore.initialState();
  const nav = jest.fn();
  const dispatch = jest.fn();

  state.profile = {
    id: 'id',
    name: 'name',
    username: 'username',
    phoneNumber: '',
    email: '',
    isMigratory: false,
    avatarExt: 'png',
    lastUpdate: 123123,
  };
  useContext.mockReturnValueOnce({
    state: state,
    dispatch: dispatch,
  });

  const component = render(<EditProfile navigation={{navigate: nav}} />);
  await fireEvent.press(component.getByTestId('editAvatarBtn'));

  expect(launchImageLibrary.mock.calls[0][0]).toStrictEqual({
    mediaType: 'photo',
    includeBase64: true,
    maxWidth: 400,
    maxHeight: 400,
  });

  await launchImageLibrary.mock.calls[0][1]({base64: undefined, type: 'type'});
  expect(dispatch).not.toBeCalledWith(GlobalStore.setLoading(true));
});

it('Test clicks (negative)', () => {
  const state = GlobalStore.initialState();
  const nav = jest.fn();
  state.profile = {
    id: 'id',
    name: 'name',
    username: 'username',
    phoneNumber: 'phoneNumber',
    email: 'email',
    isMigratory: false,
    avatarExt: 'png',
    lastUpdate: 123123,
  };
  useContext.mockReturnValueOnce({
    state: state,
    dispatch: jest.fn(),
  });

  const component = render(<EditProfile navigation={{navigate: nav}} />);
  fireEvent.press(component.getByText(StringUtils.texts.edit.profile.phoneTitle));
  expect(nav).not.toBeCalledWith('EditPhoneNumber');

  fireEvent.press(component.getByText(StringUtils.texts.edit.profile.emailTitle));
  expect(nav).not.toBeCalledWith('EditEmail');
});
