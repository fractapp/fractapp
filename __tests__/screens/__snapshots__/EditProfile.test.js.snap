// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view (empty) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="editAvatarBtn"
  >
    <Image
      height={120}
      source={
        Object {
          "uri": "uri",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 60,
          "height": 120,
          "justifyContent": "center",
          "marginTop": 20,
          "width": 120,
        }
      }
      width={120}
    />
  </View>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "onClick": [Function],
            "placeholder": "edit.profile.name_placeholder",
            "title": "edit.profile.name_title",
            "value": "",
          },
          Object {
            "onClick": [Function],
            "placeholder": "edit.profile.username_placeholder",
            "title": "edit.profile.username_title",
            "value": "",
          },
          Object {
            "onClick": [Function],
            "placeholder": "edit.profile.phone_placeholder",
            "title": "edit.profile.phone_title",
            "value": "",
          },
          Object {
            "onClick": [Function],
            "placeholder": "edit.profile.email_placeholder",
            "title": "edit.profile.email_title",
            "value": "",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#CCCCCC",
                  "marginTop": 30,
                  "paddingBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
              >
                edit.profile.name_title
              </Text>
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "marginTop": 6,
                  }
                }
              >
                edit.profile.name_placeholder
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#CCCCCC",
                  "marginTop": 30,
                  "paddingBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
              >
                edit.profile.username_title
              </Text>
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "marginTop": 6,
                  }
                }
              >
                edit.profile.username_placeholder
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#CCCCCC",
                  "marginTop": 30,
                  "paddingBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
              >
                edit.profile.phone_title
              </Text>
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "marginTop": 6,
                  }
                }
              >
                edit.profile.phone_placeholder
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#CCCCCC",
                  "marginTop": 30,
                  "paddingBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
              >
                edit.profile.email_title
              </Text>
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "marginTop": 6,
                  }
                }
              >
                edit.profile.email_placeholder
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`Test view (full) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="editAvatarBtn"
  >
    <Image
      height={120}
      source={
        Object {
          "uri": "uri",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 60,
          "height": 120,
          "justifyContent": "center",
          "marginTop": 20,
          "width": 120,
        }
      }
      width={120}
    />
  </View>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "onClick": [Function],
            "placeholder": "edit.profile.name_placeholder",
            "title": "edit.profile.name_title",
            "value": "name",
          },
          Object {
            "onClick": [Function],
            "placeholder": "edit.profile.username_placeholder",
            "title": "edit.profile.username_title",
            "value": "@username",
          },
          Object {
            "onClick": [Function],
            "placeholder": "edit.profile.phone_placeholder",
            "title": "edit.profile.phone_title",
            "value": "phoneNumber",
          },
          Object {
            "onClick": [Function],
            "placeholder": "edit.profile.email_placeholder",
            "title": "edit.profile.email_title",
            "value": "email",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#CCCCCC",
                  "marginTop": 30,
                  "paddingBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
              >
                edit.profile.name_title
              </Text>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "marginTop": 6,
                  }
                }
              >
                name
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#CCCCCC",
                  "marginTop": 30,
                  "paddingBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
              >
                edit.profile.username_title
              </Text>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "marginTop": 6,
                  }
                }
              >
                @username
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#CCCCCC",
                  "marginTop": 30,
                  "paddingBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
              >
                edit.profile.phone_title
              </Text>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "marginTop": 6,
                  }
                }
              >
                phoneNumber
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#CCCCCC",
                  "marginTop": 30,
                  "paddingBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#888888",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
              >
                edit.profile.email_title
              </Text>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "marginTop": 6,
                  }
                }
              >
                email
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
